Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/BaseRepository.cs

```diff
@@ -191,6 +191,37 @@ public async Task<IEnumerable<T>> GetAllAsync(string partition = null, int? coun
             }
         }
 
+        /// <summary>
+        /// Get paged data entities from the table storage in a partition.
+        /// </summary>
+        /// <param name="partition">Partition key value.</param>
+        /// <param name="count">The max number of desired entities.</param>
+        /// <returns>All data entities and continuation token.</returns>
+        public async Task<(IEnumerable<T>, TableContinuationToken)> GetByCountAsync(string partition = null, int? count = null)
+        {
+            var partitionKeyFilter = this.GetPartitionKeyFilter(partition);
+            var query = new TableQuery<T>().Where(partitionKeyFilter);
+            query.TakeCount = count;
+            TableQuerySegment<T> seg = await this.Table.ExecuteQuerySegmentedAsync<T>(query, null);
+            return (seg.Results, seg.ContinuationToken);
+        }
+
+        /// <summary>
+        /// Get data entities from the table storage in a partition using token.
+        /// </summary>
+        /// <param name="token">Continuation token.</param>
+        /// <param name="partition">Partition key value.</param>
+        /// <param name="count">The max number of desired entities.</param>
+        /// <returns>All data entities.</returns>
+        public async Task<(IEnumerable<T>, TableContinuationToken)> GetByTokenAsync(TableContinuationToken token, string partition = null, int? count = null)
```

Comment:
I think we can combine this w/ GetByCount into a GetPagedAsync(partition=null, count=null, token=null)

Reply:
Merged the functions.